src/main/java/dev/lambdaurora/aurorasdeco/block/entity/BlackboardBlockEntity.java

```diff
@@ -254,11 +254,20 @@ public void readNbt(NbtCompound nbt) {
 		super.readNbt(nbt);
 		this.readBlackBoardNbt(nbt);
 		this.lastUser = null;
+		this.rebuildMesh();
```

Comment:
This will crash on server!!!!

Please create a readClientNbt or similar named method and call it from readNbt only on client (environment check)

Reply:
Whoops...
I'm sorry, I tested in multiplayer, but not on a dedicated server. Thanks for catching this.
I will change that, and also test on a dedicated server.

Reply:
This shouldn't work....

What in hell Java

Reply:
I triple-checked because I was wondering if I did something wrong when testing, but it works without crashing on the dedicated server. I also verified that the readNbt method gets called. I'm not completely sure why it shouldn't work.